### PR TITLE
docs: add notes about new build pipeline to migration guide

### PR DESCRIPTION
**Pull request by myandrienko**
1. Adds a section about the new build pipeline to the v12 migration guide
2. Fixes a couple of typos

### CODE REVIEW

**Review by oliverlaz — APPROVED** [review]

File: docusaurus/docs/React/release-guides/upgrade-to-v12.mdx [hunk 1]
```diff
@@ -61,7 +61,7 @@ Replace the render functions in your custom components with container components
 Until now, it was possible to import two stylesheets as follows:
 
 ```
-import 'stream-chat-react/dist/css/v2/index.css';
+import 'stream-chat-react/dist/css/v1/index.css';
```
Comment (myandrienko) [inline docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:64]: @MartinCupela I'm not sure if this is just a typo, or it was supposed to be something else completely :) I'm going to merge this, but please take a look.

Comment (oliverlaz) [reply docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:64]: If I'm not mistaken, this import is wrong. cc: @MartinCupela

Comment (MartinCupela) [reply docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:64]: Yes, that one is the victim of changing imports.

File: docusaurus/docs/React/release-guides/upgrade-to-v12.mdx [hunk 2]
```diff
@@ -246,6 +246,25 @@ Import type alternatives if necessary.
...
 
 :::important
-**Action optional**<br/>
+**Action optional**  
 Provide custom `TypingIndicator` through the `Channel` prop.
 :::
+
+## Updated browser target
+
+Version 12 targets browsers that support ES2020. In particular, the code includes `async` functions, optional chaining (`?.`), and nullish coalescing (`??`). These features have been supported by all major desktop and mobile browsers for years, so it made sense for us to raise the baseline.
```
Comment (oliverlaz) [inline docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:255]: On a second thought, maybe it is good if we have a list of browser versions that have full es2020 support.

File: docusaurus/docs/React/release-guides/upgrade-to-v12.mdx (hunk 1 above)
Comment (oliverlaz) [reply docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:64]: What's the correct version now? :)

Comment (MartinCupela) [reply docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:64]: I will reformulate it a bit in a separate PR

**Pull request merged**